router/core/header_rule_engine_test.go

```diff
@@ -48,6 +49,43 @@ func TestNamedPropagateHeaderRule(t *testing.T) {
 	assert.Empty(t, updatedClientReq.Header.Get("X-Test-2"))
 }
 
+func TestReNamedPropagateHeaderRule(t *testing.T) {
```

Comment:
Please add another test case when `matching` is used instead of `named`.

Reply:
when we use ```matching```  instead of  ```named``` . we will need to be set ```default```  because it will use ```default``` value. if not set then it will set  empty string.